### PR TITLE
Item name overrides, drop player instanced item support

### PR DESCRIPTION
**Pull request by 0ffz**
- Allow renamed items to avoid name migration + relevant tests
- Removing item name returns default behaviour
- We no longer support player instanced items as a result because they cause a lot of headaches with this :(
- Future support will be added in the form of a parent/prefab that's created once per player per item type

### CODE REVIEW

File: geary-papermc-tracking/src/main/kotlin/com/mineinabyss/geary/papermc/tracking/items/cache/PlayerItemCache.kt [hunk 1]
```diff
@@ -24,19 +21,12 @@ abstract class PlayerItemCache<T>(
...
-        if (playerInstanced.hasInstance(entity)) {
-            playerInstanced.removeAnyKind(slot)
-        } else {
-            val pdc = entity.get<ItemStack>()?.itemMeta?.persistentDataContainer
-            if (pdc != null) entity.encodeComponentsTo(pdc)
-            entity.removeEntity()
-        }
+        val pdc = entity.get<ItemStack>()?.itemMeta?.persistentDataContainer
+        if (pdc != null) entity.encodeComponentsTo(pdc)
+        entity.removeEntity()
```
Comment (Boy0000) [inline geary-papermc-tracking/src/main/kotlin/com/mineinabyss/geary/papermc/tracking/items/cache/PlayerItemCache.kt:29]: couldnt this just be consolidated into
```kt
entity.get<ItemStack>()?.itemMeta?.persistentDataContainer?.let { entity.encodeComponentsTo(it) } ?: entity.removeEntity()
```
Is there any design reason to this? just speakin in general really

**Review by Boy0000 — APPROVED** [review]
outside of the question there I've tested this and seems to work without issues 👍 
bonfires and other items like it seem to work, which before were broken when not player_instanced

**Comment by 0ffz** [comment]
Currently working out a bug that happened here or earlier on develop where the disable interacition component isn't working anymore. I'll try to add the fix here before merging back in even if it happened outside of these changes.

File: geary-papermc-tracking/src/main/kotlin/com/mineinabyss/geary/papermc/tracking/items/cache/PlayerItemCache.kt (hunk 1 above)
Comment (0ffz) [reply geary-papermc-tracking/src/main/kotlin/com/mineinabyss/geary/papermc/tracking/items/cache/PlayerItemCache.kt:29]: I wanna keep the congnitive load down and we're doing separate things on each line so I think it's okay to have one line per step. Your example would also not remove the entity if an ItemStack was found, which we definitely always want to do :)

Comment (Boy0000) [reply geary-papermc-tracking/src/main/kotlin/com/mineinabyss/geary/papermc/tracking/items/cache/PlayerItemCache.kt:29]: true nvm i misread and thought i saw an else... but yeah sure
```kt
entity.get<ItemStack>()?.itemMeta?.persistentDataContainer?.let { entity.encodeComponentsTo(it) }
entity.removeEntity()
```

**Comment by 0ffz** [comment]
Figured out it was an issue with my local looty build being outdated :)

**Pull request merged**